### PR TITLE
Add default provider support to Az.Attestation module

### PR DESCRIPTION
**Pull request by yinqchen**
## Description

Design review:
https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/623

Added default provider support to `Az.Attestation` module
    - Added `Location` and `DefaultProvider` to `Get-AzAttestation`
    - Added `Location` and `DefaultProvider` to policy signer management cmdlets
    - Added `Location` and `DefaultProvider` to policy management cmdlets

Updated claim name from `aas-policyCertificate` to `maa-policyCertificate` for policy signer cmdlets

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_) 
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Attestation/Attestation/Commands/GetAzureAttestation.cs [hunk 1]
```diff
@@ -65,19 +68,62 @@ public class GetAzureAttestation : AttestationManagementCmdletBase
...
+            switch (ParameterSetName)
             {
-                var resourceIdentifier = new ResourceIdentifier(ResourceId);
-                Name = resourceIdentifier.ResourceName;
-                ResourceGroupName = resourceIdentifier.ResourceGroupName;
-            }
+                case DefaultProviderParameterSet:
+                    if (string.IsNullOrEmpty(Location))
+                    {
+                        this.WriteObject(AttestationClient.ListDefaultAttestation());
```
Comment (VeryEarly) [inline src/Attestation/Attestation/Commands/GetAzureAttestation.cs:103]: use
this.WriteObject(AttestationClient.ListDefaultAttestation(), true);
for collection

Comment (yinqchen) [reply src/Attestation/Attestation/Commands/GetAzureAttestation.cs:103]: Fixed.

**Comment by yinqchen** [comment]
@VeryEarly Can you please elaborate on this? 
```
switch parameter represent only two states, if it's mandatory, it means there is only one state.
You can remove this parameter
```
Remove SwitchParameter `DefaultProvider` for `Get-AzAttestation`, `Get-AzAttestationPolicy` and `Get-AzAttestationPolicySigners`?

- For `Get-AzAttestation`
  - `DefaultProvider` is mandatory for `DefaultProviderParameterSet`. Inside `DefaultProviderParameterSet`, there is another 
     optional parameter `Location`.
     If `Location` is not specified, this cmd lists all available default providers
     If `Location` is specified, this cmd lists default provider from that location
   - We need this `DefaultProvider` switch for customers to distinguish this is a regional default Provider, not a regular attestation 
     provider.
   - We will support `Get-AzAttestation` with no parameter to list all regular attestation providers from in customer's 
   subscription later

- For `Get-AzAttestationPolicy` and `Get-AzAttestationPolicySigners`
   We don't support List all default providers' policy or policy Signers. `Location` parameter is required in 
   `DefaultProviderParameterSet`. 
   We still add SwitchParameter `DefaultProvider` in `DefaultProviderParameterSet` because
    - For customers to distinguish this is a regional default Provider, not a regular attestation provider.
    - Enable same experience with `Get-AzAttestation` cmd

**Comment by VeryEarly** [comment]
> @VeryEarly Can you please elaborate on this?
> 
> ```
> switch parameter represent only two states, if it's mandatory, it means there is only one state.
> You can remove this parameter
> ```
> 
> Remove SwitchParameter `DefaultProvider` for `Get-AzAttestation`, `Get-AzAttestationPolicy` and `Get-AzAttestationPolicySigners`?
> 
> * For `Get-AzAttestation`
>   
>   * `DefaultProvider` is mandatory for `DefaultProviderParameterSet`. Inside `DefaultProviderParameterSet`, there is another
>     optional parameter `Location`.
>     If `Location` is not specified, this cmd lists all available default providers
>     If `Location` is specified, this cmd lists default provider from that location
>   * We need this `DefaultProvider` switch for customers to distinguish this is a regional default Provider, not a regular attestation
>     provider.
>   * We will support `Get-AzAttestation` with no parameter to list all regular attestation providers from in customer's
>     subscription later
> * For `Get-AzAttestationPolicy` and `Get-AzAttestationPolicySigners`
>   We don't support List all default providers' policy or policy Signers. `Location` parameter is required in
>   `DefaultProviderParameterSet`.
>   We still add SwitchParameter `DefaultProvider` in `DefaultProviderParameterSet` because
>   
>   * For customers to distinguish this is a regional default Provider, not a regular attestation provider.
>   * Enable same experience with `Get-AzAttestation` cmd

Make sense. Thanks for the explanation.

**Comment by VeryEarly** [comment]
please also suppress credentials in this file:
https://github.com/yinqchen/azure-powershell/blob/claire/tools/SecurityTools/CredScanSuppressions.json

**Comment by yinqchen** [comment]
> please also suppress credentials in this file:
> https://github.com/yinqchen/azure-powershell/blob/claire/tools/SecurityTools/CredScanSuppressions.json

Added the suppression.

**Pull request merged**